src/Compilers/CSharp/Test/Symbol/Symbols/Source/NullablePublicAPITests.cs

```diff
@@ -1679,16 +1682,76 @@ void M()
 
             var comp = CreateCompilation(source, options: WithNonNullTypesTrue(), parseOptions: featureFlagOff);
             comp.VerifyDiagnostics(
-                    // (7,16): warning CS0219: The variable 'o' is assigned but its value is never used
-                    //         object o = null;
-                    Diagnostic(ErrorCode.WRN_UnreferencedVarAssg, "o").WithArguments("o").WithLocation(7, 16),
-                    // (7,20): warning CS8600: Converting null literal or possible null value to non-nullable type.
-                    //         object o = null;
-                    Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "null").WithLocation(7, 20));
+                // (5,12): warning CS0414: The field 'C.field' is assigned but its value is never used
+                //     object field = null;
+                Diagnostic(ErrorCode.WRN_UnreferencedFieldAssg, "field").WithArguments("C.field").WithLocation(5, 12),
+                // (5,20): warning CS8625: Cannot convert null literal to non-nullable reference type.
+                //     object field = null;
+                Diagnostic(ErrorCode.WRN_NullAsNonNullable, "null").WithLocation(5, 20),
+                // (9,16): warning CS0219: The variable 'o' is assigned but its value is never used
+                //         object o = null;
+                Diagnostic(ErrorCode.WRN_UnreferencedVarAssg, "o").WithArguments("o").WithLocation(9, 16),
+                // (9,20): warning CS8600: Converting null literal or possible null value to non-nullable type.
+                //         object o = null;
+                Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "null").WithLocation(9, 20));
 
             Assert.False(comp.NullableSemanticAnalysisEnabled);
         }
 
+        private class CSharp73ProvidesNullableSemanticInfo_Analyzer : DiagnosticAnalyzer
+        {
+            public int HitCount;
+
+            private static readonly DiagnosticDescriptor Descriptor =
+               new DiagnosticDescriptor("XY0000", "Test", "Test", "Test", DiagnosticSeverity.Warning, true, "Test", "Test");
+
+            public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
+            => ImmutableArray.Create(Descriptor);
+
+            public override void Initialize(AnalysisContext context)
+            {
+                context.RegisterSyntaxNodeAction(AnalyzeMemberAccess, SyntaxKind.SimpleMemberAccessExpression);
+            }
+
+            private void AnalyzeMemberAccess(SyntaxNodeAnalysisContext context)
+            {
+                var node = (MemberAccessExpressionSyntax)context.Node;
+                var model = context.SemanticModel;
+                var info = model.GetTypeInfo(node.Expression);
+
+                Assert.Equal(PublicNullableAnnotation.Annotated, info.Nullability.Annotation);
+                Assert.Equal(PublicNullableFlowState.MaybeNull, info.Nullability.FlowState);
+
```

Comment:
Did you confirm that this assert fails for implementation from iteration 2?

Reply:
> In the old iteration, this assert does not fail, because we never reach it. Instead an assert in MemberSemanticModel fails:

Could you please confirm that, if you step over that debug assert in the debugger (usually, I am able to continue after debugger breaks execution due to a debug assert while a test is being debugged), then the test (I assume at least one of these asserts) fails the "normal" x-unit way? 

---
In reply to: [463241933](https://github.com/dotnet/roslyn/pull/46402#discussion_r463241933) [](ancestors = 463241933)

Reply:
Alternatively, you can execute release flavor of this test. Whichever you find easier.

---
In reply to: [463296735](https://github.com/dotnet/roslyn/pull/46402#discussion_r463296735) [](ancestors = 463296735,463241933)

Reply:
Thanks for the suggestion to run in Release mode. With that, I find that the Xunit assert failed within the `AnalyzeMemberAccess` method via an unexpected AAD0001 diagnostic:

```console
[xUnit.net 00:00:01.92]     Microsoft.CodeAnalysis.CSharp.UnitTests.NullablePublicAPITests.CSharp73ProvidesNullableSemanticInfo [FAIL]
  X Microsoft.CodeAnalysis.CSharp.UnitTests.NullablePublicAPITests.CSharp73ProvidesNullableSemanticInfo [863ms]
  Error Message:

Expected:
Actual:
    Diagnostic("AD0001").WithArguments("Microsoft.CodeAnalysis.CSharp.UnitTests.NullablePublicAPITests+CSharp73ProvidesNullableSemanticInfo_Analyzer", "Xunit.Sdk.EqualException", "Assert.Equal() Failure
Expected: Annotated
Actual:   None", "Exception occurred with following context:
Compilation: 638fbd72-f58d-40c4-9ddf-c62ce4a9d737
SyntaxTree:
SyntaxNode: s.ToString [MemberAccessExpressionSyntax]@[90..100) (7,12)-(7,22)

Xunit.Sdk.EqualException: Assert.Equal() Failure
Expected: Annotated
Actual:   None
   at Xunit.Assert.Equal[T](T expected, T actual, IEqualityComparer`1 comparer) in C:\projects\xunit\src\xunit.assert\Asserts\EqualityAsserts.cs:line 41
   at Xunit.Assert.Equal[T](T expected, T actual) in C:\projects\xunit\src\xunit.assert\Asserts\EqualityAsserts.cs:line 25
   at Microsoft.CodeAnalysis.CSharp.UnitTests.NullablePublicAPITests.CSharp73ProvidesNullableSemanticInfo_Analyzer.AnalyzeMemberAccess(SyntaxNodeAnalysisContext context) in C:\Users\rikki\src\roslyn\src\Compilers\CSharp\Test\Symbol\Symbols\Source\NullablePublicAPITests.cs:line 1723
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.<>c__60`1.<ExecuteSyntaxNodeAction>b__60_0(ValueTuple`2 data) in C:\Users\rikki\src\roslyn\src\Compilers\Core\Portable\DiagnosticAnalyzer\AnalyzerExecutor.cs:line 880
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.ExecuteAndCatchIfThrows_NoLock[TArg](DiagnosticAnalyzer analyzer, Action`1 analyze, TArg argument, Nullable`1 info) in C:\Users\rikki\src\roslyn\src\Compilers\Core\Portable\DiagnosticAnalyzer\AnalyzerExecutor.cs:line 1569
```